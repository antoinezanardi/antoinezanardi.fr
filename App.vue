<template>
  <div class="app-content">
    <MyProfile/>
    <AboutMe/>
    <MySkills/>
    <MyPortfolio/>
    <MyExperience/>
    <MyEducation/>
  </div>
</template>

<script lang="ts" setup>
import aos from "aos";
import { AboutMe, MyProfile, MySkills, MyPortfolio, MyExperience, MyEducation } from "#components";
import { onMounted, useNuxtApp } from "#imports";

const { $bootstrap } = useNuxtApp();

onMounted((): void => {
  aos.init();
  const tooltipTriggerList = document.querySelectorAll("[data-bs-toggle=\"tooltip\"]");
  tooltipTriggerList.forEach((tooltip: Element) => new $bootstrap.Tooltip(tooltip));
});
</script>