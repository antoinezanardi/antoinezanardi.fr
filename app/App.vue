<template>
  <div>
    <NavBar/>

    <div class="app-content">
      <MyProfile/>

      <AboutMe/>

      <MySkills/>

      <MyPortfolio/>

      <MyExperience/>

      <MyEducation/>

      <PageFooter/>
    </div>
  </div>
</template>

<script lang="ts" setup>
import aos from "aos";

import { AboutMe, MyProfile, MySkills, MyPortfolio, MyExperience, MyEducation, PageFooter, NavBar } from "#components";
import { onMounted, useNuxtApp, useHead } from "#imports";

const { t } = useI18n();
const { $bootstrap } = useNuxtApp();

useHead({
  title: "Antoine ZANARDI",
  meta: [{ name: "description", content: t("App.meta.description") }],
  htmlAttrs: { lang: "fr" },
});

onMounted((): void => {
  aos.init();
  const tooltipTriggerList = document.querySelectorAll("[data-bs-toggle=\"tooltip\"]");
  tooltipTriggerList.forEach((tooltip: Element) => new $bootstrap.Tooltip(tooltip));
});
</script>