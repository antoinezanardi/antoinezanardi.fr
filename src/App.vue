<template>
  <div id="app">
    <Navbar/>
    <div class="page-content">
      <Profile/>
      <About/>
      <Skills/>
      <Portfolio/>
      <Experience/>
      <Education/>
<!--      <Contact/>-->
    </div>
    <Footer/>
  </div>
</template>

<script>
import Navbar from "./components/Navbar";
import Profile from "./components/Profile";
import About from "./components/About";
import Skills from "./components/Skills";
import Portfolio from "./components/Portfolio";
import Experience from "./components/Experience";
import Education from "./components/Education";
import Footer from "./components/Footer";

export default {
    name: "app",
    components: {
        Footer,
        Education,
        Experience,
        Portfolio,
        Skills,
        About,
        Profile,
        Navbar,
    },
};
</script>

<style></style>