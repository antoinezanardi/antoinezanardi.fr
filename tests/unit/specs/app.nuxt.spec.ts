import type { mount } from "@vue/test-utils";
import type { ComponentMountingOptions } from "@vue/test-utils/dist/mount";
import { stampInHtml } from "dev-stamp";

import App from "@/App.vue";
import { mountSuspendedComponent } from "@tests/unit/utils/helpers/mount.helpers";

vi.mock("dev-stamp");

describe("App Component", () => {
  let wrapper: ReturnType<typeof mount<typeof App>>;

  async function mountAppComponent(options: ComponentMountingOptions<typeof App> = {}): Promise<ReturnType<typeof mount<typeof App>>> {
    return mountSuspendedComponent(App, { ...options });
  }

  beforeEach(async() => {
    wrapper = await mountAppComponent();
  });

  it("should match snapshot when rendered.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should use head for sso purposes when rendered.", () => {
    const expectedUseHeadAttribute = {
      title: "Antoine ZANARDI",
      meta: [{ name: "description", content: "App.meta.description" }],
      htmlAttrs: { lang: "fr" },
    };

    expect(useHead).toHaveBeenCalledExactlyOnceWith(expectedUseHeadAttribute);
  });

  it("should stamp a comment in HTML when rendered.", () => {
    const expectedComment = "ğŸ‘‹ Hey ! J'ai aussi crÃ©Ã© ğŸ’® `dev-stamp` qui a gÃ©nÃ©rÃ© ce message ! Retrouvez le sur GitHub ou npm !";

    expect(stampInHtml).toHaveBeenCalledExactlyOnceWith(expectedComment);
  });
});