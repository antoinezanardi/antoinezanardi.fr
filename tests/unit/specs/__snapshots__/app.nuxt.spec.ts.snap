// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Component > should match snapshot when rendered. 1`] = `
"<div>
  <nav-bar-stub></nav-bar-stub>
  <main class="app-content">
    <my-profile-stub></my-profile-stub>
    <about-me-stub></about-me-stub>
    <my-skills-stub></my-skills-stub>
    <my-portfolio-stub></my-portfolio-stub>
    <my-experience-stub></my-experience-stub>
    <my-education-stub></my-education-stub>
    <page-footer-stub></page-footer-stub>
  </main>
</div>"
`;
